### PR TITLE
remove PILOT_SKIP_VALIDATE_TRUST_DOMAIN

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

This flag was introduced in #16798, not valid anymore

### CODE REVIEW

**Review by GregHanson — CHANGES_REQUESTED** [review]
a release note should be added to highlight the removal of `PILOT_SKIP_VALIDATE_TRUST_DOMAIN` env var so users know it no longer needs to be configured in their deployments

**Comment by howardjohn** [comment]
We have some usage of this in our distro, can you give me a bit of time to check with the owners if we can remove it? I think we can just want to verify.

**Comment by shankgan** [comment]
looks LGTM to me

**Comment by howardjohn** [comment]
LGTM

**Review by richardwxn — APPROVED** [review]

**Review by linsun — CHANGES_REQUESTED** [review]
Can you pls specify the alternative and document it properly before it is removed?

I am very sure there are users who use it.

**Comment by linsun** [comment]
Is the better solution to use https://istio.io/latest/docs/reference/config/istio.mesh.v1alpha1/#MeshConfig-CertificateData? @hzxuzhonghu

File: releasenotes/notes/43424.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+apiVersion: release-notes/v2
+kind: feature
+area: security
+
+releaseNotes:
+  - |
+    **Removed** `PILOT_SKIP_VALIDATE_TRUST_DOMAIN` env from pilot which was unused for a long time.
```
Comment (dhawton) [inline releasenotes/notes/43424.yaml:7]:
```suggestion
    **Removed** `PILOT_SKIP_VALIDATE_TRUST_DOMAIN` environment variable from Pilot.
```

Also recommend an alternative suggestion.

**Review by dhawton — CHANGES_REQUESTED** [review]

**Comment by hzxuzhonghu** [comment]
There is no alternative, the env is totally useless since very long before

**Comment by linsun** [comment]
> There is no alternative, the env is totally useless since very long before

Can I request this to be discussed in Wed's WG meeting before approval of the PR?

**Comment by hzxuzhonghu** [comment]
sure

**Comment by danehans** [comment]
@hzxuzhonghu you will also need to sign the [CLA](https://github.com/istio/istio/pull/43424/checks?check_run_id=11478714629) for your PR to pass CI.

File: releasenotes/notes/43424.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+apiVersion: release-notes/v2
+kind: feature
+area: security
+
+releaseNotes:
+  - |
+    **Removed** `PILOT_SKIP_VALIDATE_TRUST_DOMAIN` environment variable from Pilot.
```
Comment (danehans) [inline releasenotes/notes/43424.yaml:7]: Needs newline.

**Comment by liminw** [comment]
This flag was added to allow cross-trust-domain communication. From the description, it seems that this flag is no longer used? I wonder how cross-trust-domain communication is enabled today. @howardjohn @linsun Any idea?

**Comment by linsun** [comment]
> This flag was added to allow cross-trust-domain communication. From the description, it seems that this flag is no longer used? I wonder how cross-trust-domain communication is enabled today. @howardjohn @linsun Any idea?

@liminw you are asking the right Q! this is exactly why we brought up this topic in today's WG meeting. @costinm suggested to use TrustDomainAlias instead of CertificateData for single root multiple trust domain validation.  What is your thoughts?

**Comment by hzxuzhonghu** [comment]
It was introduced to control filter setting, and the filter was removed. why such an useless env should be kept? 

```
	filterConfig := &authn_filter.FilterConfig{
		Policy: p,
		// we can always set this field, it's no-op if mTLS is not used.
		SkipValidateTrustDomain: features.SkipValidateTrustDomain.Get(),
	}
```

**Comment by howardjohn** [comment]
the important part is https://github.com/istio/istio/blob/1c70653e0eec55aed47674590767cd2dd4e8ff64/pilot/pkg/networking/plugin/authn/util.go#L24

**Comment by liminw** [comment]
This flag is to 

> It was introduced to control filter setting, and the filter was removed. why such an useless env should be kept?
> 
> ```
> 	filterConfig := &authn_filter.FilterConfig{
> 		Policy: p,
> 		// we can always set this field, it's no-op if mTLS is not used.
> 		SkipValidateTrustDomain: features.SkipValidateTrustDomain.Get(),
> 	}
> ```

@hzxuzhonghu Thanks for the info. Could you point me to the PR where the filter was removed? Thanks.

**Comment by hzxuzhonghu** [comment]
sorry for mistaken, it is set to always skip in  #26224

**Pull request closed without merging**